package/kedro_viz/models/graph.py

```diff
@@ -497,10 +508,57 @@ def __post_init__(self, data_node: DataNode):
             with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
                 self.plot = json.load(fs_file)
 
+        if data_node.is_metric_node():
+            if not dataset._exists() or self.filepath is None:
+                return
+
+            import pandas as pd
+            from kedro.extras.datasets.tracking.metrics_dataset import MetricsDataSet
+
+            dataset = cast(MetricsDataSet, dataset)
+            load_path = get_filepath_str(dataset._get_load_path(), dataset._protocol)
+            with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
+                self.metrics = json.load(fs_file)

```

Comment:
I think if you replace these lines with just

```python
self.metrics = dataset.load()
```

it will work the same way, i.e. always load the latest version of the dataset, and it's more reliable because it's native Kedro API. We couldn't use it for plotly because plotly was asymmetrical.

Reply:
I get the error -  Loading not supported for `MetricsDataSet`

Reply:
@limdauto after our discussions we decided not to implement the `load` on metrics dataset.
https://github.com/quantumblacklabs/private-kedro/blob/master/kedro/extras/datasets/tracking/metrics_dataset.py#L107-L108

Reply:
Yeah, I got that error.